spec.md

```diff
@@ -797,6 +797,12 @@ The ZKP protocols use nonces which MUST be different in each context.
 
 BBS signatures can be implemented on any pairing-friendly curve. Using BLS12-381 the signature achieves close to 128-bit security, and is the recommended curve at this time.
 
+## Nonce selection
+
+The proofs generated in this specification insures that the key material was applied on the specified nonce. A verifier-specified cryptographically random nonce provides strong protections against replay attacks. In some settings, proofs can be generated in a non-interactive fashion; in this case verifiers MUST be able to verify the uniqueness of the nonce values (provers could use a unique verifier ID combined with a timestamp to help with this).
+
+The proof generation acts a digital signature on the nonce, which could be used to design BBS+ based digital signature mechanisms. In this case, the nonce could be the hash digest of the message to sign, and the resulting proofs (potentially containing disclosed attribute values) would form the signature.
```

Comment:
While I share the idea that there are very good use cases where deterministic nonce can be used here, I think they make more sense in higher level protocols. Especially, since the wording used here may incorrectly guide implementors to adopt timestamp for a nonce.
I think a shorter guidance describing that this parameter be unique, and random enough would be more appropriate.

Specific implementations may still make a conscious choice to use deterministic value where security or risk impact tolerates this. For example: attendees at a conference can prepare a proof by using the room number for a conference where both verifier and presenter know this nonce and require no interaction, thus simplifying the verification experience.

Reply:
Agreed that we need clearer usage "profiles" to scope the implementation choices. Really interested in defining these.